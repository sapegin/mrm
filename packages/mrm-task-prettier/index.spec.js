jest.mock('fs');
jest.mock('mrm-core/src/util/log', () => ({
	added: jest.fn(),
}));
jest.mock('mrm-core/src/npm', () => ({
	install: jest.fn(),
}));

const { install } = require('mrm-core');
const { getTaskOptions } = require('mrm');
const vol = require('memfs').vol;
const task = require('./index');

const stringify = o => JSON.stringify(o, null, '  ');

const packageJson = stringify({
	name: 'unicorn',
	scripts: {
		test: 'jest',
	},
});

afterEach(() => {
	vol.reset();
	install.mockClear();
});

it('should add Prettier', async () => {
	vol.fromJSON({
		'/package.json': packageJson,
	});

	task(await getTaskOptions(task));

	expect(vol.toJSON()).toMatchSnapshot();
	expect(install).toBeCalledWith({ prettier: '>=2' });
});

it('should use a custom indent', async () => {
	vol.fromJSON({
		'/package.json': packageJson,
	});

	task(await getTaskOptions(task, false, { indent: 'tab' }));

	expect(vol.toJSON()['/.prettierrc']).toMatchSnapshot();
});

it('should use a custom options', async () => {
	vol.fromJSON({
		'/package.json': packageJson,
	});

	task(
		await getTaskOptions(task, false, { prettierOptions: { printWidth: 333 } })
	);

	expect(vol.toJSON()['/.prettierrc']).toMatchSnapshot();
});

it('should use a custom pattern', async () => {
	vol.fromJSON({
		'/package.json': packageJson,
	});

	task(
		await getTaskOptions(task, false, { prettierPattern: '**/*.{xxx,yyy}' })
	);

	expect(vol.toJSON()['/package.json']).toMatchSnapshot();
});

it('should use custom overrides', async () => {
	vol.fromJSON({
		'/package.json': packageJson,
	});

	task(
		await getTaskOptions(task, false, {
			prettierOverrides: [
				{
					files: '*.css',
					options: {
						printWidth: 42,
					},
				},
			],
		})
	);

	expect(vol.toJSON()['/.prettierrc']).toMatchSnapshot();
});

it('should infer options from EditorConfig', async () => {
	vol.fromJSON({
		'/.editorconfig': '[*]\nindent_style = space\nindent_size = 2',
		'/package.json': packageJson,
	});

	task(await getTaskOptions(task));

	expect(vol.toJSON()['/.prettierrc']).toMatchSnapshot();
});

it('should extend a custom pattern defined in a package.json script', async () => {
	vol.fromJSON({
		'/package.json': stringify({
			name: 'unicorn',
			scripts: {
				format: "prettier --write '**/*.{xxx,yyy}'",
				test: 'jest',
			},
		}),
	});

	task(await getTaskOptions(task));

	expect(vol.toJSON()['/package.json']).toMatchSnapshot();
});
