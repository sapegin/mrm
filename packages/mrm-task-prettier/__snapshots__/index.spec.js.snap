// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add Prettier 1`] = `
Object {
  "/.prettierrc": "{
  \\"useTabs\\": true,
  \\"overrides\\": [
    {
      \\"files\\": \\"*.md\\",
      \\"options\\": {
        \\"printWidth\\": 70,
        \\"useTabs\\": false,
        \\"trailingComma\\": \\"none\\",
        \\"arrowParens\\": \\"avoid\\",
        \\"proseWrap\\": \\"never\\"
      }
    }
  ]
}",
  "/package.json": "{
  \\"name\\": \\"unicorn\\",
  \\"scripts\\": {
    \\"test\\": \\"jest\\",
    \\"posttest\\": \\"npm run format\\",
    \\"format\\": \\"prettier --loglevel warn --write \\\\\\"**/*.{js,css,md}\\\\\\"\\"
  }
}",
}
`;

exports[`should extend a custom pattern defined in a package.json script 1`] = `
"{
  \\"name\\": \\"unicorn\\",
  \\"scripts\\": {
    \\"format\\": \\"prettier --loglevel warn --write \\\\\\"**/*.{xxx,yyy,js,css,md}\\\\\\"\\",
    \\"test\\": \\"jest\\",
    \\"posttest\\": \\"npm run format\\"
  }
}"
`;

exports[`should infer options from EditorConfig 1`] = `
"{
  \\"overrides\\": [
    {
      \\"files\\": \\"*.md\\",
      \\"options\\": {
        \\"printWidth\\": 70,
        \\"useTabs\\": false,
        \\"trailingComma\\": \\"none\\",
        \\"arrowParens\\": \\"avoid\\",
        \\"proseWrap\\": \\"never\\"
      }
    }
  ]
}"
`;

exports[`should use a custom indent 1`] = `
"{
  \\"useTabs\\": true,
  \\"overrides\\": [
    {
      \\"files\\": \\"*.md\\",
      \\"options\\": {
        \\"printWidth\\": 70,
        \\"useTabs\\": false,
        \\"trailingComma\\": \\"none\\",
        \\"arrowParens\\": \\"avoid\\",
        \\"proseWrap\\": \\"never\\"
      }
    }
  ]
}"
`;

exports[`should use a custom options 1`] = `
"{
  \\"printWidth\\": 333,
  \\"overrides\\": [
    {
      \\"files\\": \\"*.md\\",
      \\"options\\": {
        \\"printWidth\\": 70,
        \\"useTabs\\": false,
        \\"trailingComma\\": \\"none\\",
        \\"arrowParens\\": \\"avoid\\",
        \\"proseWrap\\": \\"never\\"
      }
    }
  ]
}"
`;

exports[`should use a custom pattern 1`] = `
"{
  \\"name\\": \\"unicorn\\",
  \\"scripts\\": {
    \\"test\\": \\"jest\\",
    \\"posttest\\": \\"npm run format\\",
    \\"format\\": \\"prettier --loglevel warn --write \\\\\\"**/*.{xxx,yyy}\\\\\\"\\"
  }
}"
`;

exports[`should use custom overrides 1`] = `
"{
  \\"useTabs\\": true,
  \\"overrides\\": [
    {
      \\"files\\": \\"*.css\\",
      \\"options\\": {
        \\"printWidth\\": 42
      }
    }
  ]
}"
`;
